test/functional/test_framework/util.py

```diff
@@ -391,7 +391,11 @@ def connect_nodes(from_connection, node_num):
     from_connection.addnode(ip_port, "onetry")
     # poll until version handshake complete to avoid race conditions
     # with transaction relaying
-    wait_until(lambda:  all(peer['version'] != 0 for peer in from_connection.getpeerinfo()))
+    # See comments in net_processing:
+    # * Must have a version message before anything else
+    # * Must have a verack message before anything else
+    wait_until(lambda: all(peer['version'] != 0 for peer in from_connection.getpeerinfo()))
+    wait_until(lambda: all(peer['bytesrecv_per_msg'].pop('verack', 0) == 24 for peer in from_connection.getpeerinfo()))
```

Comment:
It's slightly hacky to use the receive statistics here to see if the local peer has received the `verack` message and completed the handshake. But the only way to do it without RPC changes, I suppose.

Reply:
There is `fSuccessfullyConnected` or so, which should do the same thing, but yeah it is not exposed on RPC